src/visualizers-workbench/P5VisualizerTemplate.ts

```diff
@@ -0,0 +1,174 @@
+    params = {
+        /** md
+- stepSize: How far to step when the user presses a left or right arrow key.
+         **/
+        stepSize: {
+            value: this.stepSize,
+            forceType: 'integer',
+            displayName: 'Step size',
+            required: true,
+        },
+    }
+
+    // internal parameters. the value of `index` will always be overwritten in
+    // `inhabit()`, but TypeScript can't infer that, so we have to give `index`
+    // an initial value
+    index = 0
+    justStepped = false
+    flash = 0
+
+    checkParameters() {
+        const status = super.checkParameters()
+
+        // make sure the step size is positive
+        if (this.params.stepSize.value <= 0) {
+            status.isValid = false
+            status.errors.push('Step size must be positive')
+        }
+
+        return status
+    }
+
+    inhabit(element: HTMLElement): void {
+        // do the behind-the-scenes work of managing p5 canvases. this work is
+        // finicky, and your visualizer won't work unless it's done properly, so
+        // you should leave it to the superclass unless you have a very
+        // compelling reason to do it yourself
+        super.inhabit(element)
+    }
+
+    setup() {
+        // the base setup function includes the `createCanvas()` call that must
+        // appear in every p5 setup function
+        //
+        //   https://p5js.org/reference/#/p5/createCanvas
+        //
+        super.setup()
+
+        // accessing `this.sketch` triggers some consistency checks, so it's
+        // most parsimonious to access it at most once per function call,
+        // storing its value as a local constant for future reference
+        const sketch = this.sketch
+
+        // set the stroke color and text alignment, which won't change from
+        // frame to frame
+        sketch.stroke(51, 51, 255)
+        sketch.textAlign(sketch.CENTER, sketch.CENTER)
+
+        // start at the beginning of the sequence
+        this.index = this.seq.first
+    }
+
+    draw() {
+        // accessing `this.sketch` triggers some consistency checks, so it's
+        // most parsimonious to access it at most once per function call,
+        // storing its value as a local constant for future reference
+        const sketch = this.sketch
+
+        // scale and center the coordinate system. calculations involving the
+        // canvas dimensions need to be done every frame, because the dimensions
+        // can change at any time
+        sketch.translate(0.5 * sketch.width, 0.5 * sketch.height)
+        const smallDim = Math.min(sketch.width, sketch.height)
+        sketch.textSize(0.2 * smallDim)
+
+        // paint the background. for an animated visualizer, this is often done
+        // as the first drawing step in each frame, wiping out the previous
+        // frame and leaving a blank canvas to draw on
+        sketch.background(206, 212, 218)
+
```

Comment:
Same comment about colors. If you want to use RGB values, I'd suggest making them named instance variables in setup, e.g. `this.myFavoriteMauve = this.sketch.color(206, 212, 218)` and then using the symbolic names here. Or again, just use a CSS color.

Reply:
I took those color values from the page style, so I've changed the sketch to do that automatically during setup.

Reply:
Aaron and I discussed this and agreed that he would assign RGB-specified colors symbolic names in setup(), and then use them via those names in draw()